astropy/modeling/fitting.py

```diff
@@ -353,6 +357,9 @@ def __call__(self, model, x, y, z=None, weights=None, rcond=None):
             raise ModelLinearityError('Model is not linear in parameters, '
                                       'linear fit methods should not be used.')
 
+        if hasattr(model, "sunbmodel_names"):
```

Comment:
Typo here "submodel_names".

Could you add a small test to `test_fitters.py` which varifies that an error is raised in this case. It would have caught the typo.